File: tecnologias/java/maven/kata-01-legacy-code/src/test/java/org/bacraftmanship/model/ComercioTest.java
package org.bacraftmanship.model;

import org.assertj.core.data.Percentage;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class ComercioTest {

    private Comercio comercio;

    @BeforeEach
    public void setup() {
        comercio = new Comercio();
    }

    @Test
    public void comercianteCompraGaseosaPor200VendePor350LaGananciaEs100() {
        String gaseosa = comprar("coca", 200, 1);

        Double renta = vender(gaseosa, 350);

        assertThat(renta).isCloseTo(150, Percentage.withPercentage(0.02));
    }

    @Test
    public void comercianteCompra100GaseosaPor500VendePor100LaGananciaEs95() {
        String gaseosa = comprar("coca", 500, 100);

        Double renta = vender(gaseosa, 100);

        assertThat(renta).isCloseTo(95, Percentage.withPercentage(0.02));
    }

    private Double vender(String gaseosa, double precioVenta) {
        return comercio.vender(gaseosa, precioVenta);
    }

    private String comprar(String gaseosa, double precioCompra, int cantidad) {
        return comercio.compra(gaseosa, precioCompra, cantidad);
    }
}

File: tecnologias/kotlin/gradle/bac_kata_01_legacy_code/src/test/kotlin/org/bacraftmanship/model/ComercioTest.kt
package org.bacraftmanship.model

import org.assertj.core.api.Assertions
import org.assertj.core.data.Percentage
import org.junit.Before
import org.junit.Test

class ComercioTest {
    private var comercio: Comercio? = null

    @Before
    fun setup() {
        comercio = Comercio()
    }

    @Test
    fun comercianteCompraGaseosaPor200VendePor350LaGananciaEs100() {
        val gaseosa = comprar("coca", 200.0, 1)
        val renta = vender(gaseosa, 350.0)
        Assertions.assertThat(renta)
            .isCloseTo(150.0, Percentage.withPercentage(0.02))
    }

    @Test
    fun comercianteCompra100GaseosaPor500VendePor100LaGananciaEs95() {
        val gaseosa = comprar("coca", 500.0, 100)
        val renta = vender(gaseosa, 100.0)
        Assertions.assertThat(renta)
            .isCloseTo(95.0, Percentage.withPercentage(0.02))
    }

    private fun vender(gaseosa: String, precioVenta: Double): Double? {
        return comercio!!.vender(gaseosa, precioVenta)
    }

    private fun comprar(gaseosa: String, precioCompra: Double, cantidad: Int): String {
        return comercio!!.compra(gaseosa, precioCompra, cantidad)
    }
}